bodo/io/arrow_reader.h

```diff
@@ -543,6 +545,40 @@ class ArrowReader {
     bool reported_init_stage_metrics = false;
     bool reported_read_stage_metrics = false;
 
+    const int32_t batch_readahead = init_batch_readahead();
+    const int32_t frag_readahead = init_frag_readahead();
+
+    /*
+     * @brief determine how many batches to read ahead. Arrows defaults are
+     * based on a single reader and we need to adjust for the number of ranks on
+     * the node while always ensuring at least one batch is read ahead.
+     */
+    int32_t init_batch_readahead() {
+        if (!this->parallel) {
+            // If not parallel, we don't need to adjust the readahead
+            return arrow::dataset::kDefaultBatchReadahead;
+        }
+        int ranksOnNode = std::get<0>(dist_get_ranks_on_node());
+        return std::max(arrow::dataset::kDefaultBatchReadahead / ranksOnNode,
+                        1);
```

Comment:
Do we need to reduce batch readahead at all? The default is 8M rows which is independent of file size and row groups.
https://github.com/apache/arrow/blob/3f95bc18828d3649fe0c4a437bd70150819952cb/cpp/src/arrow/dataset/scanner.h#L52

Reply:
Ok, doesn't matter if performance is ok in general.